qi-lib/on.rkt

```diff
@@ -21,28 +22,29 @@
    #`((flow clause) #,@(syntax->list #'ags))])
 
 (define-syntax-parser flow-lambda
-  [(_ (arg:id ...) expr:expr ...)
-   #'(lambda (arg ...)
-       (on (arg ...)
-           expr ...))]
-  [(_ rest-args:id expr:expr ...)
-   #'(lambda rest-args
-       (on (rest-args)
-           expr ...))])
+  [(_ args:formals clause:clause)
+   #:with ags (attribute args.params)
+   #'(lambda args
+       (on ags
+           clause))])
 
 (define-alias π flow-lambda)
 
 (define-syntax-parser define-flow
-  [(_ (name:id arg:id ...) expr:expr)
+  [(_ (head . args:formals) clause:clause)
+   #'(define head
+       (flow-lambda args
+         clause))]
+  [(_ name:id clause:clause)
    #'(define name
-       (flow-lambda (arg ...)
-         expr))]
-  [(_ name:id expr:expr)
-   #'(define name
-       (flow expr))])
+       (flow clause))])
 
 (define-syntax-parser let/flow
-  [(_ ([var:id val:expr] ...) body ...)
-   #'(let ([var val] ...)
-       (on (var ...)
-         body ...))])
+  [(_ ([var:id val:expr] ...) clause:clause)
```

Comment:
Yeah, as @benknoble  guessed, I had been considering removing these `let` forms, since I realized that `on` and other embeddings like `~>` are already analogous to `let` for flows, in the sense that they simultaneously define + apply a function.

The only difference is that `let` would also add bindings, but as we can already do that with a normal `let`, and this may not be an idiomatic pattern for Qi anyway, it doesn't seem as though we need to provide special support for it from the language. Any reason we should keep them?

If you'd be interested in removing these forms as part of this PR, that would be fine by me!